src/lib.rs

```diff
@@ -976,6 +1004,43 @@ pub mod pallet {
             }
         }
 
+        /// Creates a new liquidity token given an asset
+        fn create_liquidity_token(
```

Comment:
I can never exist beforehand. If it exists it implies the Exchange has already been created, and a second call to the `create_exchange` extrinsic would fail. This is actually the expected behaviour.